### PR TITLE
BITG masternode support

### PR DESCRIPTION
**Pull request by konez2k**
Updated also the BitcoinGreen rebrand to BitGreen

### CODE REVIEW

**Comment by kyuupichan** [comment]
The code copy-paste isn't acceptable; use inheritance.  Also remove your .coverage file.

**Comment by Hunghus** [comment]
hello @konez2k We need fork pivx. Can you help us? We will pay you. We need urgently. Save us

**Comment by konez2k** [comment]
@kyuupichan I'll try to base it from Dash, excluding the masternode list the other commands should be compatible

**Comment by konez2k** [comment]
@Hunghus this is not the channel, discord konez2k#6060

**Comment by konez2k** [comment]
sorry @kyuupichan, I'll leave this open for 2 or 3 weeks, because BITG is changing also the masternode list.

File: electrumx/server/daemon.py [hunk 1]
```diff
@@ -496,3 +496,14 @@ def strip_mtp_data(self, raw_block):
         blocks = await self._send_vector('getblock', params_iterable)
         # Convert hex string to bytes
         return [hex_to_bytes(self.strip_mtp_data(block)) for block in blocks]
+
+
+class BitGreenDaemon(Daemon):
```
Comment (kyuupichan) [inline electrumx/server/daemon.py]: Please rename this DashDaemonBase, and have DashDaemon and SmartCashDaemon inherit it.

File: electrumx/server/session.py [hunk 2]
```diff
@@ -1676,3 +1676,137 @@ def truncate_auxpow(self, headers_full_hex, start_height):
             height += 1
 
         return headers.hex()
+
+
```
Comment (kyuupichan) [inline electrumx/server/session.py]: Similarly here - this is common to all dash derivatives so it should be DashElectrumXBase and have DashElectrumX inherit it.  Then there's no copy-paste.

**Pull request closed without merging**